executor/src/witgen/jit/processor.rs

```diff
@@ -320,8 +344,16 @@ pub struct Error<'a, T: FieldElement> {
     pub incomplete_identities: Vec<(&'a Identity<T>, i32)>,
 }
 
+#[derive(PartialEq, Eq)]
 pub enum ErrorReason {
+    /// This error means that the current branch (if it is a branch)
+    /// is actually not reachable.
+    ConflictingConstraints,
```

Comment:
Should we add the constraint that was violated, and the referenced values? Can be a later PR I think, but I imagine if the system is actually unsatisfiable, that would be helpful.

Reply:
I mean in the end, we have to extract something like an "unsat core", i.e. a minimal set of identities that lead to a conflict. If you just provide the identity that conflicted, it is of not much use. In the runtime solving, this was often something like `XIsZero * X = 0`.